be/src/exec/pipeline/fragment_executor.cpp

```diff
@@ -663,11 +663,7 @@ void FragmentExecutor::_fail_cleanup() {
         }
         if (_query_ctx->count_down_fragments()) {
             auto query_id = _query_ctx->query_id();
-            if (ExecEnv::GetInstance()->query_context_mgr()->remove(query_id)) {
-                if (_wg) {
-                    _wg->decr_num_queries();
```

Comment:
if there are outstanding queries during wg configeration is changing?

Reply:
QueryContext holds a shared_ptr of the workgroup, so this changed workgroup cannot be deleted until this QueryContext is destructed..